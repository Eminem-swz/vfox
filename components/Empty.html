<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Empty 空状态 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.82adce24.css">
    <link rel="modulepreload" href="/vfox/assets/Home.e3398b0e.js">
    <link rel="modulepreload" href="/vfox/assets/app.9b8f0faa.js">
    <link rel="modulepreload" href="/vfox/assets/components_Empty.md.42720013.lean.js">
    <link rel="modulepreload" href="/vfox/assets/app.9b8f0faa.js">
    <meta name="twitter:title" content="Empty 空状态 | Vfox for Vue3">
    <meta property="og:title" content="Empty 空状态 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-675d8756><div class="sidebar-button" data-v-675d8756><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-675d8756 data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-675d8756></div><div class="nav" data-v-675d8756><nav class="nav-links" data-v-675d8756 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-83e92a68><nav class="nav-links nav" data-v-83e92a68 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-83e92a68><!--[--><li class="sidebar-link"><p class="sidebar-link-item">基础</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Button">Button</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Icon">Icon</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Image">Image</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Drawer">Drawer</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dropdown">Dropdown</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">反馈</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dialog">Dialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SwipeCell">SwipeCell</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">数据输入</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Form">Form</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Input">Input</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Radio">Radio</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Checkbox">Checkbox</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Slider">Slider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Switch">Switch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stepper">Stepper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Calendar">Calendar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SearchBar">SearchBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NumberKeyboard">NumberKeyboard</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImageUploader">ImageUploader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Range">Range</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Rate">Rate</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">导航</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/BackTop">BackTop</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NavBar">NavBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopMenu">PopMenu</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tab">Tab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SideTab">SideTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabBar">TabBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabView">TabView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Sticky">Sticky</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollTab">ScrollTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/IndexView">IndexView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Fixed">Fixed</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Pagination">Pagination</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">展示</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Layout">Layout</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cell">Cell</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Group">Group</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Collapse">Collapse</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/components/Empty">Empty</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#import">Import</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#props">Props</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#slots">Slots</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#底部（-default）">底部（#default）</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#图片区域（image）">图片区域（image）</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Skeleton">Skeleton</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Swiper">Swiper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImagePreview">ImagePreview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/FlatList">FlatList</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NoticeBar">NoticeBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Divider">Divider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Badge">Badge</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Progress">Progress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tag">Tag</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Order">Order</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountDown">CountDown</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Steps">Steps</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Price">Price</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Avatar">Avatar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Timeline">Timeline</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stopwatch">Stopwatch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Result">Result</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/LoadMore">LoadMore</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TimeAgo">TimeAgo</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountUp">CountUp</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CircleProgress">CircleProgress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActivityIndicator">ActivityIndicator</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">其他</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Copy">Copy</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div><iframe style="width:375px;height:667px;margin-top:1.5rem;" src="https://godxiaoji.github.io/vfox/demo/#/Empty" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="empty-空状态" tabindex="-1">Empty 空状态 <a class="header-anchor" href="#empty-空状态" aria-hidden="true">#</a></h1><h2 id="import" tabindex="-1">Import <a class="header-anchor" href="#import" aria-hidden="true">#</a></h2><div class="language-"><pre><code>import { Empty } from &#39;vfox&#39;
</code></pre></div><p>具体的引入方式可以参考<a href="./../#引入组件">引入组件</a>。</p><h2 id="props" tabindex="-1">Props <a class="header-anchor" href="#props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>description</td><td>string</td><td></td><td>否</td><td>描述文字</td></tr><tr><td>type</td><td>string</td><td>&#39;default&#39;</td><td>否</td><td>图片类型，可选 &#39;default&#39;, &#39;error&#39;, &#39;network&#39;, &#39;search&#39;</td></tr></tbody></table><h2 id="slots" tabindex="-1">Slots <a class="header-anchor" href="#slots" aria-hidden="true">#</a></h2><h3 id="底部（-default）" tabindex="-1">底部（#default） <a class="header-anchor" href="#底部（-default）" aria-hidden="true">#</a></h3><div class="language-"><pre><code>&lt;fx-empty description=&quot;网络错误&quot; type=&quot;network&quot;&gt;
  &lt;fx-button&gt;刷新试试&lt;/fx-button&gt;
&lt;/fx-empty&gt;
</code></pre></div><h3 id="图片区域（image）" tabindex="-1">图片区域（image） <a class="header-anchor" href="#图片区域（image）" aria-hidden="true">#</a></h3><div class="language-"><pre><code>&lt;fx-empty description=&quot;描述文字&quot;&gt;
  &lt;template #image&gt;
    &lt;fx-image src=&quot;xxx.jpg&quot; /&gt;
  &lt;/template&gt;
&lt;/fx-empty&gt;
</code></pre></div><p>注：设置了该 <code>slot</code> 则不展示内置图片</p></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><a class="link" href="/vfox/components/Collapse" data-v-38ede35f><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-38ede35f><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-38ede35f>Collapse</span></a></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/components/Skeleton" data-v-38ede35f><span class="text" data-v-38ede35f>Skeleton</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_drawer.md\":\"6d8cc285\",\"apis_modal.md\":\"47141a7d\",\"apis_notify.md\":\"c1b38e37\",\"apis_popover.md\":\"59134bc3\",\"apis_preview.md\":\"cbb69585\",\"apis_toast.md\":\"96d27e50\",\"apis_index.md\":\"4a90e465\",\"components_actionsheet.md\":\"188ae6ff\",\"components_activityindicator.md\":\"021ee5b9\",\"components_avatar.md\":\"362f02b1\",\"components_backtop.md\":\"235bd461\",\"components_badge.md\":\"05d688d8\",\"components_button.md\":\"c98c5005\",\"components_calendar.md\":\"94f2b57a\",\"components_cascader.md\":\"90ea6e6a\",\"components_cell.md\":\"d5d623a7\",\"components_checkbox.md\":\"3d0d090c\",\"components_circleprogress.md\":\"d6baf2f5\",\"components_collapse.md\":\"94dad2f7\",\"components_copy.md\":\"c1f9c219\",\"components_countdown.md\":\"2dca531e\",\"components_countup.md\":\"bcdb61a2\",\"components_datepicker.md\":\"45805008\",\"components_dialog.md\":\"5ae31140\",\"components_divider.md\":\"9e476f98\",\"components_drawer.md\":\"c55ed563\",\"components_dropdown.md\":\"f62182c0\",\"components_empty.md\":\"42720013\",\"components_fixed.md\":\"2bacfadc\",\"components_flatlist.md\":\"595e0d0b\",\"components_form.md\":\"57d02dc7\",\"components_group.md\":\"78cba411\",\"components_icon.md\":\"84c03668\",\"components_image.md\":\"8f27216d\",\"components_imagepreview.md\":\"93dededf\",\"components_imageuploader.md\":\"3bb5e289\",\"components_indexview.md\":\"938bb626\",\"components_input.md\":\"eee52426\",\"components_layout.md\":\"05ef0b6d\",\"components_loadmore.md\":\"122041cb\",\"components_modal.md\":\"7042bff6\",\"components_navbar.md\":\"e48d00e3\",\"components_noticebar.md\":\"0b9fcd43\",\"components_notify.md\":\"35c6d7ac\",\"components_numberkeyboard.md\":\"316075d8\",\"components_order.md\":\"a56630ef\",\"components_pagination.md\":\"826cb220\",\"components_picker.md\":\"28928b78\",\"components_popdialog.md\":\"04137b2a\",\"components_popmenu.md\":\"609d46cd\",\"components_popover.md\":\"f65d2426\",\"components_price.md\":\"9d09a6e2\",\"components_progress.md\":\"62019cc3\",\"components_radio.md\":\"741acea9\",\"components_range.md\":\"49b1f9ec\",\"components_rate.md\":\"a41c3b62\",\"components_result.md\":\"b9cf5cd9\",\"components_scrolltab.md\":\"1a2528e1\",\"components_scrollview.md\":\"111d89a7\",\"components_searchbar.md\":\"77d8e293\",\"components_sidetab.md\":\"de6fc489\",\"components_skeleton.md\":\"39e2b819\",\"components_slider.md\":\"48af82ff\",\"components_stepper.md\":\"49053488\",\"components_steps.md\":\"18ff8530\",\"components_sticky.md\":\"6db5d4c0\",\"components_stopwatch.md\":\"07d49320\",\"components_swipecell.md\":\"90b48e0e\",\"components_swiper.md\":\"e4f77bcc\",\"components_switch.md\":\"633ddddb\",\"components_tab.md\":\"700c23d8\",\"components_tabbar.md\":\"52a86aea\",\"components_tabview.md\":\"9abfebcc\",\"components_tag.md\":\"ae2e477b\",\"components_timeago.md\":\"83942205\",\"components_timeline.md\":\"8d80ff53\",\"components_toast.md\":\"66403d92\",\"components_index.md\":\"72126489\",\"design_color.md\":\"d5ff546b\",\"design_index.md\":\"4e9f0647\",\"index.md\":\"220ab970\"}")</script>
    <script type="module" async src="/vfox/assets/app.9b8f0faa.js"></script>
    
  </body>
</html>