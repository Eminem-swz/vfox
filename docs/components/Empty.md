# Empty 空状态

## Props

| 属性        | 类型   | 默认值    | 必填 | 说明                                                      |
| ----------- | ------ | --------- | ---- | --------------------------------------------------------- |
| description | string |           | 否   | 描述文字                                                  |
| type        | string | 'default' | 否   | 图片类型，可选 'default' / 'error' / 'network' / 'search' |

## CSS

| 属性               | 默认值  | 说明   |
| ------------------ | ------- | ------ |
| --empty-font-color | #8c8c8c | 文本色 |

## Slots

### 底部（#default）

```
<fx-empty description="网络错误" type="network">
  <fx-button>刷新试试</fx-button>
</fx-empty>
```

### 图片区域（image）

```
<fx-empty description="描述文字">
  <template #image>
    <fx-image src="xxx.jpg" />
  </template>
</fx-empty>
```

注：设置了该 `slot` 则不展示内置图片
