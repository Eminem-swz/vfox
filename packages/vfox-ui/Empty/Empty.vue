<template>
  <div class="fx-empty">
    <slot name="image" v-if="$slots.image"></slot>
    <template v-else>
      <img class="fx-empty_image" :src="imageUrl" />
    </template>
    <p class="fx-empty_description" v-show="description">
      {{ description }}
    </p>
    <slot></slot>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue'
import type { PropType } from 'vue'
import { createEnumsValidator, getEnumsValue } from '../helpers/validator'
import type { EmptyType } from './types'
import { images } from './images'

const TYPE_NAMES: EmptyType[] = ['default', 'error', 'network', 'search']

export default defineComponent({
  name: 'fx-empty',
  props: {
    // 描述文字
    description: {
      type: String,
      default: ''
    },
    // 类型
    type: {
      type: String as PropType<EmptyType>,
      validator: createEnumsValidator(TYPE_NAMES),
      default: TYPE_NAMES[0]
    }
  },
  setup(props) {
    const imageUrl = computed(() => {
      return images[getEnumsValue(TYPE_NAMES, props.type)]
    })

    return {
      imageUrl
    }
  }
})
</script>
