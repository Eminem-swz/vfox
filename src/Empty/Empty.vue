<template>
  <div class="fx-empty">
    <slot name="image" v-if="$slots.image"></slot>
    <template v-else>
      <img class="fx-empty_image" :src="imageUrl" />
    </template>
    <p class="fx-empty_description" v-show="description">
      {{ description }}
    </p>
    <slot></slot>
  </div>
</template>

<script>
import { inArray } from '../helpers/util'

const TYPE_NAMES = [
  'default',
  'error',
  'network',
  'search',
  'permission',
  'service'
]

export default {
  name: 'fx-empty',
  props: {
    // 描述文字
    description: {
      type: String,
      default: ''
    },
    // 类型
    type: {
      validator(val) {
        return inArray(val, TYPE_NAMES)
      },
      default: TYPE_NAMES[0]
    }
  },
  computed: {
    imageUrl() {
      return `https://cdn.fox2.cn/vfox/empty/${
        inArray(this.type, TYPE_NAMES) ? this.type : TYPE_NAMES[0]
      }@3x.png`
    }
  }
}
</script>
