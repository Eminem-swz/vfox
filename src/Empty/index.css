.fx-empty {
  --empty-font-color: #8c8c8c;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 24px 0;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}

.fx-empty_image {
  width: 160px;
  height: 160px;
}

.fx-empty_description {
  color: var(--empty-font-color);
  text-align: center;
  word-break: break-all;
  word-wrap: break-word;
  margin: 16px 0 0 0;
  font-size: 14px;
  line-height: 19.6px;
}

.fx-empty_description + * {
  margin-top: 16px;
}
