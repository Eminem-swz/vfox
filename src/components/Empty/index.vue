<template>
  <div :class="[prefix + '-empty']">
    <slot class="image" v-if="hasImage"></slot>
    <template v-else>
      <img
        v-if="type === 'default'"
        :class="[prefix + '-empty_image']"
        src="../../images/empty/default@2x.png"
      />
    </template>
    <p :class="[prefix + '-empty_description']" v-show="description">
      {{ description }}
    </p>
    <slot></slot>
  </div>
</template>

<script>
import { SDKKey } from '../../config'
import { inArray } from '../../helpers/util'

const TYPE_NAMES = ['default', 'error', 'network', 'search']

export default {
  name: SDKKey + '-empty',
  components: {},
  props: {
    // 描述文字
    description: {
      type: String,
      default: ''
    },
    // 类型
    type: {
      validator(val) {
        return inArray(val, TYPE_NAMES)
      },
      default: TYPE_NAMES[0]
    }
  },
  data() {
    return { prefix: SDKKey, hasImage: false }
  },
  watch: {},
  computed: {
  },
  created() {},
  mounted() {
    if (this.$scopedSlots.image) {
      this.hasImage = true
    }
  },
  beforeDestroy() {},
  methods: {}
}
</script>

<style lang="scss">
@import '../component.module.scss';

.#{$prefix}-empty {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 24px 0;
  box-sizing: border-box;

  &_image {
    width: 224px;
    height: 180px;
  }

  &_description {
    color: $font-color;
    text-align: center;
    word-break: break-all;
    word-wrap: break-word;
    margin: 16px 0 0 0;
    font-size: 17px;
    line-height: 23.8px;
  }
}
</style>
