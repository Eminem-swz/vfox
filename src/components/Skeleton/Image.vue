<template>
  <div :class="[prefix + '-skeleton-image', { animated: useAnimation }]"></div>
</template>

<script>
import { SDKKey } from '../../config'

export default {
  name: SDKKey + '-skeleton-image',
  components: {},
  inject: {
    skeletonSubOptions: {
      default: null
    }
  },
  props: {
    // 是否显示动画
    animated: {
      type: Boolean,
      default: false
    }
  },
  data() {
    return { prefix: SDKKey }
  },
  watch: {},
  computed: {
    useAnimation() {
      if (this.skeletonSubOptions) {
        return this.skeletonSubOptions.animated
      }
      return this.animated
    }
  },
  created() {},
  mounted() {},
  beforeDestroy() {},
  methods: {}
}
</script>

<style lang="scss">
@import '../component.module.scss';

.#{$prefix}-skeleton-image {
  display: block;
  width: 100px;
  height: 100px;
  border-radius: 2px;
  @include skeleton-background;

  &.animated {
    animation: #{$prefix}-skeleton-loading 1.4s ease infinite;
  }
}
</style>
