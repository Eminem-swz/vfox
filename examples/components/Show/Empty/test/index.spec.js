/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Empty from '@/Empty'
import Image from '@/Image'

describe('Empty', () => {
  test('should render prop set description correctly', () => {
    const wrapper = mount(Empty, {
      propsData: {
        description: 'description'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render set image slot correctly', () => {
    const wrapper = mount(Empty, {
      propsData: {
        description: 'description'
      },
      slots: {
        image: mount(Image, {
          propsData: {
            src: 'https://cdn.fox2.cn/vfox/swiper/different-1.jpg'
          }
        }).html()
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
