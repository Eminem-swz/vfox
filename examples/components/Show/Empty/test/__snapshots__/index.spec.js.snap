// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty should render prop set description correctly 1`] = `
<div class="fx-empty"><img class="fx-empty_image" src="https://cdn.fox2.cn/vfox/empty/default@2x.png">
  <p class="fx-empty_description">description</p>
</div>
`;

exports[`Empty should render set image slot correctly 1`] = `
<div class="fx-empty">
  <div class="fx-image">
    <!--v-if--><i class="fx-image_loading"><svg class="fx-icon">
        <use xlink:href="#icon-ImageOutlined"></use>
      </svg></i>
    <!--v-if-->
    <!--v-if-->
  </div>
  <p class="fx-empty_description">description</p>
</div>
`;
